### PR TITLE
SQL: [docs] Add documentation for COALESCE

### PR DESCRIPTION
**Pull request by matriv**
Follows: #35253

### CODE REVIEW

File: docs/reference/sql/functions/conditional.asciidoc [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+[[sql-functions-conditional-coalesce]]
+==== `COALESCE`
+
+.Synopsis
+[source, sql]
+----
+COALESCE ( expression<1>, expression<2>, ... )
+----
+
```
Comment (astefan) [inline docs/reference/sql/functions/conditional.asciidoc]: Enumerating the parameters should happen in a section called `Input`. Also, there has to be an `Output` section. Something like:

```
==== `COALESCE`

.Synopsis:
[source, sql]
--------------------------------------------------
COALESCE ( expression<1>, expression<2>, ... )
--------------------------------------------------

*Input*:

<1> 1st expression
<2> 2nd expression
....

*Output*: one expression or `null`

.Description:

Returns the first of its arguments that is not null.
If all arguments are null, then it returns `null`.
```

Comment (astefan) [reply docs/reference/sql/functions/conditional.asciidoc]: Also, I would add in the description (to enforce the `...` in the parameters list), that the function can take an arbitrary number of arguments.

**Review by astefan — COMMENTED** [review]
Left two comments.

**Comment by MirAkram** [comment]
mj1265849@gmail.com
در تاریخ ۲۰ نوامبر ۲۰۱۸ ۷:۵۷ بعدازظهر، "Andrei Stefan" <
notifications@github.com> نوشت:

File: docs/reference/sql/functions/conditional.asciidoc (hunk 1 above)
Comment (astefan) [reply docs/reference/sql/functions/conditional.asciidoc]: The statement about arbitrary number of arguments is there. Please, ignore my comment.

**Comment by matriv** [comment]
@astrfan Thank you! pushed fixup.

**Comment by matriv** [comment]
@astefan  Thank you! pushed fixup.

**Review by astefan — APPROVED** [review]
LGTM

**Review by costin — APPROVED** [review]
LGTM

**Comment by matriv** [comment]
Backported to `6.x` with 2f19570f1b97e26320255a805ef17d9ce9d54b74

**Pull request merged**